src/coreclr/src/vm/threadsuspend.cpp

```diff
@@ -6091,18 +6091,24 @@ BOOL Thread::HandledJITCase(BOOL ForTaskSwitchIn)
             // the method returns an object reference, so we know whether to protect
             // it or not.
             EECodeInfo codeInfo(ip);
-            VOID *pvHijackAddr = GetHijackAddr(this, &codeInfo);
+
+            GcInfoDecoder gcInfoDecoder(codeInfo.GetGCInfoToken(), DECODE_REVERSE_PINVOKE_VAR);
```

Comment:
Can we move this into `EECodeManager::GetReturnKind` so that we crack the GCInfo just once?

It should also fix the x86 build break that I expect you will see with this change.

Reply:
Just so I am following this correctly. Your suggestion would seem to imply that `GetHijackAddr()` return `NULL` or perhaps some other out argument to indicate not this is a reverse P/Invoke? I guess alternatively, the current GCInfo in this method could be plumbed down to the `EECodeManager::GetReturnKind` call.

Reply:
Basically, I am replacing the EECodeManager::GetReturnKind by `bool GetReturnAddressHijackInfo(GCInfoToken gcInfoToken, ReturnKind * returnKind)` that returns false if hijacking is not possible / true if it is. And if it is possible, it also sets the returnKind output parameter. This new method combines the original EECodeManager::GetReturnKind with the checking of the reverse pinvoke frame flag.